src/cets_discovery.erl

```diff
@@ -0,0 +1,146 @@
+    start_common(start_link, Opts).
+
+start_common(F, Opts) ->
+    Args =
+        case Opts of
+            #{name := Name} ->
+                [{local, Name}, ?MODULE, Opts, []];
+            _ ->
+                [?MODULE, Opts, []]
+        end,
+    apply(gen_server, F, Args).
+
+add_table(Server, Table) ->
+    gen_server:call(Server, {add_table, Table}).
+
+get_tables(Server) ->
+    gen_server:call(Server, get_tables).
+
+info(Server) ->
+    {ok, Tables} = get_tables(Server),
+    [cets:info(Tab) || Tab <- Tables].
+
+-spec init(term()) -> {ok, state()}.
+init(Opts) ->
+    Mod = maps:get(backend_module, Opts, cets_discovery_file),
+    self() ! check,
+    Tables = maps:get(tables, Opts, []),
+    BackendState = Mod:init(Opts),
+    {ok, #{results => [], tables => Tables,
+           backend_module => Mod, backend_state => BackendState,
+           timer_ref => undefined}}.
+
+-spec handle_call(term(), from(), state()) -> {reply, term(), state()}.
+handle_call({add_table, Table}, _From, State = #{tables := Tables}) ->
+    case lists:member(Table, Tables) of
+        true ->
+            {reply, {error, already_added}, State};
+        false ->
+            State2 = State#{tables := [Table | Tables]},
+            {reply, ok, handle_check(State2)}
+    end;
+handle_call(get_tables, _From, State = #{tables := Tables}) ->
+    {reply, {ok, Tables}, State};
+handle_call(Msg, From, State) ->
+    ?LOG_ERROR(#{what => unexpected_call, msg => Msg, from => From}),
+    {reply, {error, unexpected_call}, State}.
+
+-spec handle_cast(term(), state()) -> {noreply, state()}.
+handle_cast(Msg, State) ->
+    ?LOG_ERROR(#{what => unexpected_cast, msg => Msg}),
+    {noreply, State}.
+
+-spec handle_info(term(), state()) -> {noreply, state()}.
+handle_info(check, State) ->
+    {noreply, handle_check(State)};
+handle_info(Msg, State) ->
+    ?LOG_ERROR(#{what => unexpected_info, msg => Msg}),
+    {noreply, State}.
+
+terminate(_Reason, _State) ->
+    ok.
+
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+-spec handle_check(state()) -> state().
+handle_check(State = #{tables := []}) ->
+    %% No tables to track, skip
+    schedule_check(State);
+handle_check(State = #{backend_module := Mod, backend_state := BackendState}) ->
+    {Res, BackendState2} = Mod:get_nodes(BackendState),
+    State2 = handle_get_nodes_result(Res, State),
+    schedule_check(State2#{backend_state := BackendState2}).
+
+handle_get_nodes_result({error, _Reason}, State) ->
+    State;
+handle_get_nodes_result({ok, Nodes}, State = #{tables := Tables}) ->
+    Results = [do_join(Tab, Node) || Tab <- Tables, Node <- Nodes, node() =/= Node],
+    report_results(Results, State),
+    State#{results := Results}.
+
+schedule_check(State) ->
+    cancel_old_timer(State),
+    TimerRef = erlang:send_after(5000, self(), check),
+    State#{timer_ref := TimerRef}.
+
+cancel_old_timer(#{timer_ref := OldRef}) when is_reference(OldRef) ->
+    _ = erlang:cancel_timer(OldRef),
```

Comment:
prevents from dialyzer warning.

Reply:
Which option triggers this warning? I see a lot of them enabled. Sometimes it might be better to disable a check if it forces us to write unnecessary code.